source/engine/graphics/Renderer.cpp

```diff
@@ -59,10 +91,13 @@ Renderer::~Renderer() noexcept
     m_renderPasses.reset();
 }
 
-void Renderer::Record(Commands* commands, const PerFrameRenderState& state, const MeshStorage& meshStorage, uint32_t currentSwapChainImageIndex)
+void Renderer::Record(PerFrameGpuContext* currentFrame, const PerFrameRenderState& state, const MeshStorage& meshStorage, uint32_t currentSwapChainImageIndex)
 {
+    auto cmd = currentFrame->CommandBuffer();
+
     OPTICK_CATEGORY("Renderer::Record", Optick::Category::Rendering);
```

Comment:
Try to keep optick macros at the very top of functions so they don't miss anything